### PR TITLE
Route updates

### PR DESCRIPTION
**Pull request by adamwoodcock98**
## Before you start

1. Have you run `git pull origin main` on your branch to bring in the latest changes?
2. Is there an issue on the repository for this request? If not, please create one.
3. Have you followed the correct workflow in the [team documentation](https://docs.google.com/document/d/13yi709ax4nmfTG9J0F57yEFpYRH7fdtczwvVJCrNIsM/edit#)?
4. Conflict resolution to be dealt with **on GitHub**.

## What do these code changes do?
These changes allow the user to navigate to their profile from the drop-down menu. The user can now search for new friends, view their profile and add them as a friend.

## How do these code changes affect the application?
There is now more data flow around the app, finally allowing access to the profile through the UI as opposed to direct URLs.

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have created a PR ticket on the [Trello board](https://trello.com/b/P8lEsvZe/acebook).
- [x] I will link the PR to the Trello card once created.
- [x] I will send the PR in the Slack channel and ping @here.

### CODE REVIEW

**Comment by r94o** [comment]
[I would like to send them a friend request](https://trello.com/c/7CLlv1rL/35-i-would-like-to-send-them-a-friend-request)

File: cypress/integration/user_can_view_profile_spec.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ describe.only("Viewing a profile", () => {
     cy.get(".about-container").should("contain", "About me")
   });
 
-  it("should display the users posts", () => {
+  xit("should display the users posts", () => {
```
Comment (clovellbsc) [inline cypress/integration/user_can_view_profile_spec.js:16]: and this one?

File: cypress/integration/user_can_view_profile_spec.js [hunk 2]
```diff
@@ -28,4 +28,46 @@ describe.only("Viewing a profile", () => {
...
+    cy.signUp();
+    cy.login();
+
+    cy.get("#dropdownUser1").click();
+    cy.get("#dropdown-profile-link").click();
+    cy.get("#add-friend-button").should("not.exist");
+
+    cy.get("#dropdownUser1").click();
+    cy.get("#sign-out-button").click();
+    cy.signUp("Dick", "Dick-Dickinson", "dick@dong.com", "secured1ck");
```
Comment (clovellbsc) [inline cypress/integration/user_can_view_profile_spec.js:41]: maybe change this? 🤨

File: cypress/integration/user_can_view_profile_spec.js [hunk 3]
```diff
@@ -28,4 +28,46 @@ describe.only("Viewing a profile", () => {
...
+
+    cy.get("#searchBox").type("Dick Dick-Dickinson");
+    cy.get("#searchButton").click();
+    cy.get(".user-container").click();
+
+    cy.get("#add-friend-button").should("exist");
+  });
+
+  it("should add a friend and list the friend on the users profile", () => {
+    cy.signUp("Dick", "Dick-Dickinson", "dick@dong.com", "secured1ck");
```
Comment (clovellbsc) [inline cypress/integration/user_can_view_profile_spec.js:52]: same here

**Review by clovellbsc — CHANGES_REQUESTED** [review]

**Review by PKilgarriff — APPROVED** [review]
Extensive feature guys, cracking work!

**Review by george-h-h — APPROVED** [review]
Love the process! Best of luck with the next incremental step of the implementation.

**Review by clovellbsc — APPROVED** [review]

**Pull request merged**